ControlFlowArraysAndStrings/ControlFlowArraysAndStrings/Program.cs

```diff
@@ -0,0 +1,343 @@
+                                "raspberry",
+                                "apricot" };
+
+            // Original Array.
+            Console.WriteLine("Original Array: ");
+            Print(fruits);
+
+            // Sort method.
+            Console.WriteLine("\nTesting the Sort method:");
+            string[] sortedFruits = Sort(fruits);
+            Print(sortedFruits);
+
+            // AddElementToStart method.
+            Console.WriteLine("\nTesting the AddElementToStart method:");
+            string[] addWatermelon = AddElementToStart(fruits, "watermelon");
+            Print(addWatermelon);
+
+            // AddElementToEnd method.
+            Console.WriteLine("\nTesting the AddElementToEnd method:");
+            string[] addKiwi = AddElementToEnd(fruits, "kiwi");
+            Print(addKiwi);
+
+            // AddElementAnywhere method.
+            Console.WriteLine("\nTesting the AddElementAnywhere method:");
+            string[] addLime = AddElementAnywhere(fruits, 3, "lime");
+            Print(addLime);
+
+            string[] addLime2 = AddElementAnywhere(fruits, 7, "lime");
+            Print(addLime2);
+
+            string[] addLime3 = AddElementAnywhere(fruits, 0, "lime");
+            Print(addLime3);
+
+            // RemoveElementFromStart method.
+            Console.WriteLine("\nTesting the RemoveElementFromStart method:");
+            string[] removeBanana = RemoveElementFromStart(fruits);
+            Print(removeBanana);
+
+            // RemoveElementFromEnd method.
+            Console.WriteLine("\nTesting the RemoveElementFromEnd method:");
+            string[] removeApricot = RemoveElementFromEnd(fruits);
+            Print(removeApricot);
+
+            // RemoveElementFromAnywhere method.
+            Console.WriteLine("\nTesting the RemoveElementFromAnywhere method:");
+            string[] removePineapple = RemoveElementFromAnywhere(fruits, 3);
+            Print(removePineapple);
+
+            string[] removeStart = RemoveElementFromAnywhere(fruits, 0);
+            Print(removeStart);
+
+            string[] removeEnd = RemoveElementFromAnywhere(fruits, 6);
+            Print(removeEnd);
+
+            // Login method.
+            Console.WriteLine("\nTesting the Login method:");
+            Login();
+        }
+
+        /// <summary>
+        /// Sorts a copy of the string array in alphabetical order.
+        /// </summary>
+        /// <param name="array">The array to be sorted.</param>
+        /// <returns>A sorted copy of the specified array.</returns>
+        public static string[] Sort(string[] array)
+        {
+            string[] sorted = Copy(array);
+
+            for (int i = 0; i < sorted.Length - 1; i++)
+            {
+                for (int j = 0; j < sorted.Length - i - 1; j++)
+                {
+                    // Check if CompareTo() returns 1, if it does, swap positions.
+                    if (sorted[j].CompareTo(sorted[j + 1]) == 1)
```

Comment:
The fact that `String.CompareTo` returns `1` or `-1` is an implementation detail.  According to the docs for the method (https://docs.microsoft.com/en-us/dotnet/api/system.string.compareto?view=net-5.0)...

```
Less than zero | This instance precedes strB.
Zero | This instance has the same position in the sort order as strB.
Greater than zero | This instance follows strB.
```

So technically it is allowed to return any number greater than 0 in this case.  You should test for that `CompareTo(...) > 0` and not `CompareTo(...) == 1` to prevent possibly getting bugs when you upgrade from one .NET version to the next.

Reply:
Oh interesting, thank you for pointing that out.